app/src/main/java/org/mozilla/tv/firefox/pocket/PocketVideoStore.kt

```diff
@@ -0,0 +1,91 @@
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.tv.firefox.pocket
+
+import android.content.Context
+import android.content.res.AssetManager
+import android.util.Log
+import androidx.annotation.AnyThread
+import org.mozilla.tv.firefox.telemetry.SentryIntegration
+
+private const val LOGTAG = "PocketVideoStore"
+
+private const val KEY_VIDEO_JSON = "video_json"
+private const val VIDEO_STORE_NAME = "Pocket-Global-Video-Recs"
+
+private const val BUNDLED_VIDEOS_PATH = "bundled/pocket_videos.json"
+
+/**
+ * The minimum number of valid videos we must receive from the server if we want to display them.
+ * This number is set to the number of videos that appear on screen at the time of writing.
+ */
+private const val REQUIRED_POCKET_VIDEO_COUNT = 4
+
+/**
+ * Saves the Pocket video recommendations as a raw JSON String and loads them in data structures for the app.
+ * Bad data should saved so bad data should never be returned.
+ *
+ * This class is thread safe.
+ */
+class PocketVideoStore(
+    appContext: Context,
+    private val assets: AssetManager,
+    private val convertJSONToPocketVideos: (String) -> List<PocketViewModel.FeedItem>?
+) {
+
+    // We use SharedPrefs because it's simple, it handles concurrency (so we don't even need to think about
+    // which threads we access this from), and the file sizes will be small.
+    private val sharedPrefs = appContext.getSharedPreferences(VIDEO_STORE_NAME, 0)
+
+    /**
+     * @return true if the save was successful, false if the save was not successful because the JSON is invalid.
+     */
+    @AnyThread
+    fun save(json: String): Boolean {
+        if (!isJSONValid(json)) {
+            return false
+        }
+
+        sharedPrefs.edit()
+            .putString(KEY_VIDEO_JSON, json)
+            .apply()
+
+        return true
+    }
+
+    private fun isJSONValid(rawJSON: String): Boolean {
+        // While we don't need the conversion result, this function already handles validation so we use
+        // it to validate the videos.
+        val convertedVideos = convertJSONToPocketVideos(rawJSON)
+        return convertedVideos != null &&
+
+            // Guarantee a minimum number of Pocket videos: e.g. if the server only returns one valid video,
+            // we wouldn't want to overwrite what the user already has to show only one video.
+            convertedVideos.size >= REQUIRED_POCKET_VIDEO_COUNT
+    }
+
+    /**
+     * @return the list of loaded videos. This should never happen but in case of error, the empty list is returned.
+     */
+    @AnyThread
+    fun load(): List<PocketViewModel.FeedItem> {
+        fun loadBundledTiles(): String = assets.open(BUNDLED_VIDEOS_PATH).bufferedReader().use { it.readText() }
+
+        val rawJSON =  sharedPrefs.getString(KEY_VIDEO_JSON, null) ?: loadBundledTiles()
+
+        val convertedVideos = convertJSONToPocketVideos(rawJSON)
+        if (convertedVideos == null) {
```

Comment:
Our failure cases (null and size is too small) should probably be consistent between save and load.

Reply:
My thinking is that this should never happen but we don't want to crash (to prevent crash loops) so we should return a reasonable value that will put us in an undefined state.

Then again, that undefined state could be crashing so I guess we should explicitly handle this, e.g. by returning null.